### PR TITLE
re-enable static analysis and fix issues

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

File: src/jitlayers.cpp [hunk 1]
```diff
@@ -189,7 +189,7 @@ static jl_callptr_t _jl_compile_codeinst(
             }
             jl_atomic_store_release(&this_code->invoke, addr);
         }
-        else if (this_code->invoke == jl_fptr_const_return_addr && !decls.specFunctionObject.empty()) {
+        else if (jl_atomic_load_relaxed(&this_code->invoke) == jl_fptr_const_return_addr && !decls.specFunctionObject.empty()) {
```
Comment (pchintalapudi) [inline src/jitlayers.cpp:192]: Do we want to merge these two loads of the invoke pointer (this + line 183) into a variable?

File: src/precompile.c [hunk 2]
```diff
@@ -331,16 +331,16 @@ static void jl_compile_all_defs(void)
...
         int do_compile = 0;
-        if (codeinst->invoke != jl_fptr_const_return) {
+        if (jl_atomic_load_relaxed(&codeinst->invoke) != jl_fptr_const_return) {
             if (codeinst->inferred && codeinst->inferred != jl_nothing &&
                 jl_ir_flag_inferred((jl_array_t*)codeinst->inferred) &&
                 !jl_ir_flag_inlineable((jl_array_t*)codeinst->inferred)) {
                 do_compile = 1;
             }
-            else if (codeinst->invoke != NULL || codeinst->precompile) {
+            else if (jl_atomic_load_relaxed(&codeinst->invoke) != NULL || jl_atomic_load_relaxed(&codeinst->precompile)) {
```
Comment (pchintalapudi) [inline src/precompile.c:343]: Maybe fuse these loads into a single one as well?

File: src/jitlayers.cpp (hunk 1 above)
Comment (vtjnash) [reply src/jitlayers.cpp:192]: The token `jl_fptr_const_return_addr` is only set during construction, so it is constant. Indeed, the intent is for this value to be only set once, so repeated reads will give the same answer or NULL.

File: src/precompile.c (hunk 2 above)
Comment (vtjnash) [reply src/precompile.c:343]: we don't expect this to run with threads, but we likely permit the compiler to fuse these anyways (the result should be the same either way, based on how we define and use this `invoke` field)

**Pull request merged**